src/utils/cplay.c

```diff
@@ -102,6 +111,98 @@ static const struct {
 };
 #define CPLAY_NUM_CODEC_IDS (sizeof(codec_ids) / sizeof(codec_ids[0]))
 
+static void init_stdin(void)
+{
+	struct termios term;
+	int ret;
+
+	if (!interactive)
+		return;
+
+	ret = tcgetattr(fileno(stdin), &term);
+	if (ret < 0) {
+		fprintf(stderr, "Unable to get terminal attributes.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	/* save previous terminal flags */
+	term_c_lflag = term.c_lflag;
+
+	/* save previous stdin flags and add O_NONBLOCK*/
+	stdin_flags = fcntl(fileno(stdin), F_GETFL);
+	if (stdin_flags < 0 || fcntl(fileno(stdin), F_SETFL, stdin_flags|O_NONBLOCK) < 0)
+		fprintf(stderr, "stdin O_NONBLOCK flag setup failed\n");
+
+	/* prepare to enter noncanonical mode */
+	term.c_lflag &= ~ICANON;
+
+	ret = tcsetattr(fileno(stdin), TCSANOW, &term);
+	if (ret < 0) {
+		fprintf(stderr, "Unable to set terminal attributes.\n");
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void done_stdin(void)
+{
+	struct termios term;
+	int ret;
+
+	if (!interactive)
+		return;
+
+	if (term_c_lflag == -1 || stdin_flags == -1)
+		return;
+
+	ret = tcgetattr(fileno(stdin), &term);
+	if (ret < 0) {
+		fprintf(stderr, "Unable to get terminal attributes.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	/* restore previous terminal attributes */
+	term.c_lflag = term_c_lflag;
+
+	ret = tcsetattr(fileno(stdin), TCSANOW, &term);
+	if (ret < 0) {
+		fprintf(stderr, "Unable to set terminal attributes.\n");
+		exit(EXIT_FAILURE);
+	}
+
+	/* restore previous stdin attributes */
+	ret = fcntl(fileno(stdin), F_SETFL, stdin_flags);
+	if (ret < 0) {
+		fprintf(stderr, "Unable to set stdin attributes.\n");
+		exit(EXIT_FAILURE);
+	}
+}
+
+static int do_pause(void)
+{
+	unsigned char chr;
+
+	if (!interactive)
+		return DO_NOTHING;
+
+	while (read(fileno(stdin), &chr, 1) == 1) {
+		switch(chr) {
+			case ' ':
+				if (is_paused) {
+					fprintf(stderr, "\r=== Resume ===\n");
+					return DO_RESUME;
+				} else {
+					fprintf(stderr, "\r=== Pause ===\n");
+					return DO_PAUSE;
+				}
+			break;
+			default:
+			break;
```

Comment:
both 'break' here are not correctly indented.

Reply:
Fixed